articles/android-rxjava/index.md

```diff
@@ -0,0 +1,127 @@
+
+#### Schedulers
+ThThreading is a crucial factor to consider when creating an application. One needs to handle operations on threads carefully. In Android, we have the main thread and UI thread, which do most of the work. It is therefore not advisable to run extra operations like network calls on them. RxJava comes in handy to manage how we deal with threads as we add observers to the application. Schedulers allow us to define where to perform actions and where to receive the data from observables. To specify where an event is to occur, we call the `subscibeOn` method and pass in the thread. We have several options to pass into the subscibeOn method. They are :
+
+##### `newThread`
+We call `Schedulers.newThread()` to specify that the task is to be done on a new thread. This creates a new thread dedicated to that specific task. However, it should be done carefully to prevent having many threads created and running at the same time. That leads to low performance and high CPU usage by the application.
+
+##### `io`
+This creates a new thread that does tasks that do not require a lot of computational power. The observable can also reuse these threads, and if none is available, it creates a new one. We call the `Schedulers.io()` to specify this scheduler to be used. It handles tasks like making network calls and file system tasks.
+
+##### `computation`
+Computation works in the same way as the io scheduler, only that we use it for CPU intensive work. However, it creates a limited number of threads according to the number of cores in the Android device. We use it for tasks like reading local databases, and should be used carefully. Due to the limited number of threads, if a job finds all the threads in use, it has to wait for them to finish the current tasks. Observables use `Schedulers.computation()` to run tasks using this scheduler.
+
+##### `mainThread`
+This is not a scheduler available in RxJava but is found in RxAndroid. It deals explicitly with Android-based threads like the UI thread and the main thread. You should, however, use it only when observing the data rather than running the task. To define where the scheduler should emit the data, we call the `observeOn()` method and pass it into the scheduler. So to observe data on the main thread, we call `observeOn(AndroidSchedulers.mainThread())`. Notice that the method resides in the AndroidSchedulers class and not the original `Schedulers` class.
+
+#### Application in Android
+In Android applications, we normally define on which thread the task will run and where to receive the data. And to get the data, we call the   `subscribe` method. It gives us either the emitted data or the throwable. The code structure resembles the one below.
+
+```Kotlin
+observable
+  .subscibeOn(Schedulers.io())
+  .observeOn(AndroidSchedulers.mainThread())
+  .subscribe(
+    { // it = emitted data},
+    { // it = throwable})
```

Comment:
The comment on this statement affects the `subscribe`'s closing bracket thus affecting the rest of the article.

Reply:
done...